### PR TITLE
Adds is_integrator_adaptive overload

### PR DESCRIPTION
**Pull request by kanav99**
@ChrisRackauckas

### CODE REVIEW

File: src/common_interface/integrator_types.jl [hunk 1]
```diff
@@ -147,6 +147,10 @@ DiffEqBase.check_error(integrator::AbstractSundialsIntegrator) =
     interpret_sundials_retcode(integrator.flag)
 
 DiffEqBase.postamble!(integrator::AbstractSundialsIntegrator) = nothing
+
+# false because this function is used in callbacks and cannot change dt forcibly
+DiffEqBase.is_integrator_adaptive(integrator::AbstractSundialsIntegrator) = false
```
Comment (ChrisRackauckas) [inline src/common_interface/integrator_types.jl:152]: requires a DiffEqBase version

Comment (ChrisRackauckas) [reply src/common_interface/integrator_types.jl:152]: I guess it doesn't technically error before that though.

Comment (kanav99) [reply src/common_interface/integrator_types.jl:152]: I got a message from one of the users about the error. I used `integrator.opts.adaptive` directly in apply callback

Comment (kanav99) [reply src/common_interface/integrator_types.jl:152]: Though it's not required with the isdefined now

**Pull request closed without merging**